CONTRIBUTORS.md

```diff
@@ -0,0 +1,99 @@
+5. PR rebased on master (`git rebase`, not `git pull`) 
+5. Commit message includes JIRA number
+6. No changes to code not directly related to your PR
+7. Includes functional/integration test
+8. Includes documentation
+
+Once you have submitted your PR please monitor it for comments/feedback. We reserve the right to close inactive PRs if
+you do not respond within 2 weeks (bear in mind you can always open a new PR if it is closed due to inactivity).
+
+Also, please remember that we do receive a fairly large amount of PRs and also have code to write ourselves, so we may
+not be able to respond to your PR immediately. Best place to ping us is on the thread you started on the dev mailing list.
+
+### Finding something to work on
+
+If you would like to contribute to Keycloak, but are not sure exactly what to work on, you can find a number of [open
+issues|https://issues.jboss.org/projects/KEYCLOAK/versions/12340167] that are awaiting contributions in the 
+[Keycloak JIRA](https://issues.jboss.org/projects/KEYCLOAK/versions/12340167).
+
+### Open a discussion on Keycloak Dev Mailing List
+
+As Keycloak is a community driven project we require contributors to send a description of what they are planning to 
+work on to the [Keycloak Dev Mailing List](https://lists.jboss.org/mailman/listinfo/keycloak-dev).
+
+We recommending starting the discussion prior to submitting your PR. Through the mailing list you can get valuable
+feedback both from the core Keycloak team as well as the wider community.
+
+### Create an issue in Keycloak JIRA
+
+Take your time to write a proper JIRA including a good summary and description. 
+
+Remember this may be the first thing a reviewer of your PR will look at to get an idea of what you are proposing 
+and it will also be used by the community in the future to find about what new features and enhancements are included in 
+new releases.
+
+Note: Keycloak Node.js Admin Client uses GitHub issues and not the Keycloak JIRA
+
+### Implementing
+
+We don't currently enforce a code style in Keycloak, but a good reference is the code style used by WildFly. 
+This can be retrieved from [Wildfly ide-configs](https://github.com/wildfly/wildfly-core/tree/master/ide-configs).
+To import formatting rules, see following [instructions](http://community.jboss.org/wiki/ImportFormattingRules).
+
+Do not format or refactor code that is not directly related to your contribution. If you do this it will significantly
+increase our effort in reviewing your PR. If you have a strong need to refactor code then submit a separate PR for the
+refactoring.
+
+Details on how to work with the codebase are provided in the GitHub repository for the individual projects. See the list
+of projects above.
+
+### Testing
+
+We strongly believe in functional and integration level testing. Unit testing is completely optional, but unit tests
+do not serve as test coverage, but rather development tools. With the use of frameworks like Arquillian we are able
```

Comment:
Maybe a comma here "like Arquillian, we are able..."

Reply:
Not 100% sure, but I don't think a comma there is correct